### DISCUSSION TITLE
Balloon-Compose wrap height issue

### DISCUSSION THREAD

**Discussion by isansc** [skydoves/Balloon#468] [Q&A] [answered]
I'm trying to use the compose version of balloon and its height seems to have some limit. It seems to be the same problem that the xml version faced in this issue: https://github.com/skydoves/Balloon/issues/148

But in my case I'm using the Compose version.

Below is the simple code I'm using based on the Medium arcticle:

`
    val context = LocalContext.current
    // ToolTip Builder:
    val builder = rememberBalloonBuilder {
        setArrowSize(16)
        setArrowColorMatchBalloon(true)
        setArrowPositionRules(ArrowPositionRules.ALIGN_ANCHOR)
//        setArrowPosition(0.5f)
        setTextColor(Color.Black)
        setWidth(BalloonSizeSpec.WRAP)
        setHeight(BalloonSizeSpec.WRAP)
        setPadding(12)
        setCornerRadius(12f)
        setBackgroundColor(White)
        setBalloonAnimation(BalloonAnimation.ELASTIC)
        setIsVisibleOverlay(true)
        setOverlayColor(Overlay)
        setBalloonHighlightAnimation(BalloonHighlightAnimation.NONE)
        setDismissWhenClicked(true)
    }

[...]

            IconButton(onClick = { /* Empty while testing */ }
            ) {

                Box(modifier = Modifier.fillMaxSize()) {

                    Balloon(
                        modifier = Modifier.wrapContentHeight(),
                        builder = builder,
                        balloonContent = {
                                Column() {
                                    Text(
                                        modifier = Modifier.fillMaxWidth().wrapContentHeight(),
                                        text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
                                        textAlign = TextAlign.Start,
                                        color = Color.Gray,
                                        fontSize = 14.sp
                                    )
                                    Row(
                                        modifier = Modifier
                                            .padding(0.dp, 4.dp)
                                    ) {
                                        Button(
                                            onClick = {
                                                Toast.makeText(
                                                    context,
                                                    "Primary!",
                                                    Toast.LENGTH_LONG
                                                )
                                                    .show()
                                            },
                                            modifier = Modifier
                                                .wrapContentHeight()
                                                .wrapContentWidth(),
                                            shape = RoundedCornerShape(size = 8.dp),
                                            colors = ButtonDefaults.buttonColors(backgroundColor = GreenPersonalization)
                                        ) {
                                            Text(
                                                modifier = Modifier
                                                    .padding(
                                                        start = 4.dp,
                                                        end = 4.dp,
                                                        top = 16.dp,
                                                        bottom = 16.dp
                                                    ),
                                                text = "Primary",
                                                color = White,
                                                textAlign = TextAlign.Center,
                                                fontSize = 18.sp,
                                            )
                                        }
                                        Spacer(modifier = Modifier.height(8.dp))
                                        Button(
                                            onClick = {
                                                Toast.makeText(
                                                    context,
                                                    "Secondary!",
                                                    Toast.LENGTH_LONG
                                                )
                                                    .show()
                                            },
                                            modifier = Modifier
                                                .wrapContentHeight()
                                                .wrapContentWidth(),
                                            border = BorderStroke(2.dp, Grey),
                                            shape = RoundedCornerShape(size = 8.dp),
                                            colors = ButtonDefaults.buttonColors(backgroundColor = White)
                                        ) {
                                            Text(
                                                modifier = Modifier
                                                    .padding(
                                                        start = 4.dp,
                                                        end = 4.dp,
                                                        top = 16.dp,
                                                        bottom = 16.dp
                                                    ),
                                                text = "Secondary",
                                                color = Grey,
                                                textAlign = TextAlign.Center,
                                                fontSize = 18.sp,
                                            )
                                        }
                                    }
                                }
                            }
                        }
                    ) { balloonWindow ->

                        if (showTooltip) {
                            balloonWindow.showAlignBottom()
                        }

                        IconButton(
                            modifier = Modifier.size(60.dp, 60.dp),
                            onClick = { balloonWindow.showAlignBottom() }
                        ) {
                            Box {
                                Image(
                                    modifier = Modifier
                                        .width(24.dp)
                                        .height(24.dp),
                                    painter = painterResource(id = R.drawable.ic_logo_svg),
                                    contentDescription = "TEST TOOLTIP",
                                    colorFilter = ColorFilter.tint(Red)
                                )
                            }
                        }
                    }
                }
            

`

**Comment by skydoves** [comment, marked as answer]
Hey @isansc, this issue has been resolved in the very recent release of 1.6.4. Thank you for the reporting!
